include/internal/sockets.h

```diff
@@ -116,14 +116,11 @@ struct servent *PASCAL getservbyname(const char *, const char *);
 # endif
 
 /*
- * Some IPv6 implementations are broken, disable them in known bad versions.
+ * Some IPv6 implementations are broken, you can disable them in known
+ * bad versions.
  */
 # if !defined(OPENSSL_USE_IPV6)
-#  if defined(AF_INET6) && !defined(NETWARE_CLIB)
-#   define OPENSSL_USE_IPV6 1
-#  else
-#   define OPENSSL_USE_IPV6 0
-#  endif
+#  define OPENSSL_USE_IPV6 1
```

Comment:
Presumably we still need `if defined(AF_INET6)`

Reply:
No, it's only needed if you don't define use_ipv6 which our config normally does now.

Reply:
I don't have the cycles to evaluate this answer. Just noticing that in this one case the patch was changing things not only for NETWARE, but rather also unconditionally enabling IPv6 even when AF_INET6 did not appear to be defined. Perhaps that's OK...

Reply:
Actually, @vdukhovni is correct.  @richsalz, *this* *is* the spot where `OPENSSL_USE_IPv6` gets defined for the rest of the source, nowhere else.  The reason for these kinda convoluted conditions is to give the builder the option to disable IPv6 support like like this:

```
./config -DOPENSSL_USE_IPV6=0
```

So the above should still read like this:

```
# if !defined(OPENSSL_USE_IPV6)
#  if defined(AF_INET6)
#   define OPENSSL_USE_IPV6 1
#  else
#   define OPENSSL_USE_IPV6 0
#  endif
# endif

Reply:
Thanks!  See https://github.com/openssl/openssl/pull/5030